lib/Router.js

```diff
@@ -315,24 +275,44 @@ Router.prototype.handle = function(request, connection) {
             paramValue = null;
         }
 
-        // Make sure numbers are numeric
-        var props = route.schema.properties || route.schema.items.properties;
-        if (props.hasOwnProperty(paramName)) {
-            switch (props[paramName].type) {
-                case 'integer':
-                    paramValue = parseInt(paramValue);
-                    break;
-
-                case 'number':
-                    paramValue = parseFloat(paramValue);
-                    break;
-            }
-        }
-
         request.setParam(paramName, paramValue);
         request.setQuery(paramName, paramValue);
     }
 
+    //Validate url params
+    request.validateAndCastParams();
+
+    //Validate passed in queries
+    if (request.isGet()) {
+        request.validateAndCastQueries();
+    }
+
+    var errors = request.getValidationErrors();
+    if (errors.length) {
+        emit.call(this, 'api:error', request, null, timeStart);
+
+        // Get the first error and make sure to get the base property name (e.g. if an array of values is needs
+        // to validate to an enum and the second one doesn't validate successfully, it returns propertyName.1 -
+        // this will shorten it just to propertyName.
+        var error = errors[0];
+        var propertyName = '';
+
+        if (error.path) {
+            propertyName = error.path.split('.')[0];
```

Comment:
May need to change this to `error.path.split(/[\.@]/)` to strip out anything after the `@` too.